Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx

```diff
@@ -1,51 +1,24 @@
-import React, { useCallback } from 'react';
-import isNil from 'lodash/isNil';
-
-import { useSubmissionsDetails } from '../../../hooks/submissions/use-submissions-details';
-import Button, { ButtonType } from '../../guidelines/buttons/Button';
+import React from 'react';
 
 import SubmissionsList, { ISubmissionsListProps } from './SubmissionsList';
 
-type IRefreshableSubmissionsListProps = ISubmissionsListProps
+interface IRefreshableSubmissionsListProps extends ISubmissionsListProps{
+    itemFunc?: () => React.ReactElement;
+}
 
 const RefreshableSubmissionsList = ({
     items,
     selectedSubmission,
     className = '',
-}: IRefreshableSubmissionsListProps) => {
-    const {
-        state: { currentSubmission },
-        actions: { getSubmissionResults },
-    } = useSubmissionsDetails();
-
-    const submissionsReloadBtnClassName = 'submissionReloadBtn';
-
-    const handleReloadClick = useCallback(async () => {
-        if (isNil(currentSubmission)) {
-            return;
-        }
-
-        const { problem: { id: problemId }, isOfficial, user: { id: userId } } = currentSubmission;
-
-        await getSubmissionResults(problemId, isOfficial, userId);
-    }, [ currentSubmission, getSubmissionResults ]);
-
-    return (
-        <>
-            <SubmissionsList
-              items={items}
-              selectedSubmission={selectedSubmission}
-              className={className}
-            />
-
-            <Button
-              onClick={handleReloadClick}
-              text="Reload"
-              type={ButtonType.secondary}
-              className={submissionsReloadBtnClassName}
-            />
-        </>
-    );
-};
-
+    itemFunc,
+}: IRefreshableSubmissionsListProps) => (
+    <>
+        <SubmissionsList
+          items={items}
+          selectedSubmission={selectedSubmission}
+          className={className}
+        />
+        {itemFunc?.()}
```

Comment:
What does this function do?

Reply:
We use itemFunc to pass JSX elements externally. I changed the name to be more understandable.